sklearn/utils/estimator_checks.py

```diff
@@ -1248,6 +1260,18 @@ def check_estimators_nan_inf(name, estimator_orig):
                     raise AssertionError(error_string_transform, estimator)
 
 
+@ignore_warnings
+def check_nonsquare_error(name, estimator_orig):
+    """Test that error is thrown when non-square data provided"""
+
+    X, y = make_blobs(n_samples=20, n_features=10)
+    estimator = clone(estimator_orig)
+
+    with pytest.raises(ValueError):
```

Comment:
please add the `match='...'` parameter, for a more specific detection of the `ValueError`